### PR TITLE
fix(react-tags-preview): remove double border under WHCM on dismissible circular InteractionTag

### LINKED ISSUE #29222: Circular interaction tag has double curve in windows high contrast mode
windows high contrast
image

light
image

Is this intentional?

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
fix #29222

Before: 
Screenshot 2023-09-22 at 13 36 25

After:
Screenshot 2023-09-22 at 13 35 16

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/01d9fdb588ee13cd881e9c053d13e8b8e54b3bcf'>01d9fdb588ee13cd881e9c053d13e8b8e54b3bcf</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tags-preview/src/components/InteractionTagPrimary/useInteractionTagPrimaryStyles.styles.ts [hunk 1]
```diff
@@ -185,6 +184,56 @@ const useRootWithSecondaryActionStyles = makeStyles({
...
+    '@media (forced-colors: active)': {
+      position: 'relative',
+      '::before': {
+        content: '""',
+        ...shorthands.borderTop(tokens.strokeWidthThin, 'solid'),
+        position: 'absolute',
+        top: '-1px',
+        left: '-1px',
+        right: '-1px',
+        bottom: '-1px',
```
Comment (layershifter) [inline packages/react-components/react-tags-preview/src/components/InteractionTagPrimary/useInteractionTagPrimaryStyles.styles.ts:202]: An amount of classes that will be generated really worries me.

Can we move something to base styles under `makeResetStyles` call?  For example this:

```
'@media (forced-colors: active)': {
      position: 'relative',
      '::before': {
        content: '""',
        ...shorthands.borderTop(tokens.strokeWidthThin, 'solid'),
        position: 'absolute',
        top: '-1px',
        left: '-1px',
        right: '-1px',
        bottom: '-1px',
```

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-tags-preview/src/components/InteractionTagPrimary/useInteractionTagPrimaryStyles.styles.ts (hunk 1 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags-preview/src/components/InteractionTagPrimary/useInteractionTagPrimaryStyles.styles.ts]: Oh I wanted to update you that I changed to use different makeResetStyles for different shape (rounded/circular), but I saw you're faster and approved. Thx 🙏

**Pull request merged**